validation-test/compiler_crashers_2/0083-rdar31163470-1.swift

```diff
@@ -0,0 +1,10 @@
+// RUN: not --crash %target-swift-frontend -primary-file %s -emit-ir
+// REQUIRES: asserts
+
+struct First<T> {}
+struct Second<T> {}
+
+struct Node<T> {
+  func create<U>() where T == First<U> { }
```

Comment:
@jckarter At some point last year you changed the mangling to drop requirements on outer generic parameters. However this is not correct, because these requirements are not necessarily introduced by the outer generic scope. They might be introduced by the innermost generic scope, like here, which results in conflicting mangling for the two overloads of `create`, which hits an assertion. Do you have any dupes of this?